services/wallet/bridge/swap_paraswap.go

```diff
@@ -0,0 +1,204 @@
+package bridge
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/status-im/status-go/account"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/rpc"
+	walletCommon "github.com/status-im/status-go/services/wallet/common"
+	"github.com/status-im/status-go/services/wallet/thirdparty/paraswap"
+	"github.com/status-im/status-go/services/wallet/token"
+	walletToken "github.com/status-im/status-go/services/wallet/token"
+	"github.com/status-im/status-go/transactions"
+)
+
+var zeroAddress = common.Address{}
+
+type SwapTxArgs struct {
+	transactions.SendTxArgs
+	ChainID uint64 `json:"chainId"`
+}
+
+type SwapParaswap struct {
+	paraswapClient *paraswap.ClientV5
+	priceRoute     paraswap.Route
+	transactor     *transactions.Transactor
+}
+
+func NewSwapParaswap(rpcClient *rpc.Client, transactor *transactions.Transactor, tokenManager *walletToken.Manager) *SwapParaswap {
+	return &SwapParaswap{
+		paraswapClient: paraswap.NewClientV5(walletCommon.EthereumMainnet),
+		transactor:     transactor,
+	}
+}
+
+func (s *SwapParaswap) Name() string {
+	return "Paraswap"
+}
+
+func (s *SwapParaswap) Can(from, to *params.Network, token *walletToken.Token, toToken *walletToken.Token, balance *big.Int) (bool, error) {
+	if token == nil || toToken == nil {
+		return false, errors.New("token and toToken cannot be nil")
+	}
+
+	if from.ChainID != to.ChainID {
+		return false, nil
+	}
+
+	s.paraswapClient.SetChainID(from.ChainID)
+
+	searchForToken := token.Address == zeroAddress
+	searchForToToken := toToken.Address == zeroAddress
+	if searchForToToken || searchForToken {
+		tokensList, err := s.paraswapClient.FetchTokensList(context.Background())
+		if err != nil {
+			return false, err
+		}
+
+		for _, t := range tokensList {
+			if searchForToken && t.Symbol == token.Symbol {
+				token.Address = common.HexToAddress(t.Address)
+				token.Decimals = t.Decimals
+				if !searchForToToken {
+					break
+				}
+			}
+
+			if searchForToToken && t.Symbol == toToken.Symbol {
+				toToken.Address = common.HexToAddress(t.Address)
+				toToken.Decimals = t.Decimals
+				if !searchForToken {
+					break
+				}
+			}
+		}
+	}
+
+	if token.Address == zeroAddress || toToken.Address == zeroAddress {
+		return false, errors.New("cannot resolve token/s")
+	}
+
+	return true, nil
+}
+
+func (s *SwapParaswap) CalculateFees(from, to *params.Network, token *token.Token, amountIn *big.Int, nativeTokenPrice, tokenPrice float64, gasPrice *big.Float) (*big.Int, *big.Int, error) {
+	return big.NewInt(0), big.NewInt(0), nil
```

Comment:
There is no fees from paraswap itself, like a provider fee?

Reply:
Not sure, this is from their documentation: https://developers.paraswap.network/smart-contracts/augustus-swapper#fee-structure

`ParaSwap currently doesn't charge any fees from its end-users directly, except in the case of positive slippage.`

Reply:
I guess the what you will receive is also explicit about fees